src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -120,12 +121,26 @@ case class CsvRelation protected[spark] (
       schemaFields: Seq[StructField],
       projection: MutableProjection,
       row: GenericMutableRow): Iterator[Row] = {
-    iter.map { line =>
-      val tokens = CSVParser.parse(line, csvFormat).getRecords.head
-      for (index <- 0 until schemaFields.length) {
-        row(index) = tokens.get(index)
+    iter.flatMap { line =>
+      try {
+        val records = CSVParser.parse(line, csvFormat).getRecords
```

Comment:
to be specific, I was referring to this:

```scala
CSVParser.parse(line, csvFormat).getRecords.map { tokens =>
  var index = 0
  while (index < schemaFields.length) {
    row(index) = tokens.get(index)
    index = index + 1
  }
  (projection(row))
}
```

As this is already merged, though, we can do this some time later.